### ISSUE TITLE
If `pip install -r requirements.txt` fails during an agent run, agent run should be aborted 

### ISSUE THREAD

**Issue by andyk** [agentos-project/agentos#364]
currently, a component does not check if virtualenv setup succeeds or fails since when I try to run papag, I'm seeing the `pip install -r requirements.txt` fail since i didn't have `cmake` installed on my machine. Currently the agent run command just ignores that the virtualenv requirement installation failed and tries to run the agent anyway, resulting in a confusing/misleading error message.

**Comment by andyk** [comment]
@nickjalbert just FYI i hit this. it's a small-ish thing since you can deduce that the pip install of requirements failed by reading error output of the run command anyway, but as it is, it takes a little bit more cognitive energy to figure that out

**Comment by nickjalbert** [comment]
Yeah, would be good to fix!  Will wait until it's prioritized (or until I need something small to do)
